src/health/health.controller.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright © 2021 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Controller, Get } from '@nestjs/common';
+import {HealthCheckService, HealthCheck, HttpHealthIndicator} from '@nestjs/terminus';
+import {TokensService} from "../tokens/tokens.service";
+import {basicAuth} from "../utils";
+
+@Controller('health')
+export class HealthController {
+
+    constructor(
+        private health: HealthCheckService,
+        private http: HttpHealthIndicator,
+        private readonly tokensService: TokensService
+    ) {}
+
+    @Get('/liveness')
+    @HealthCheck()
+    liveness() {
+        return this.health.check([]);
+    }
+
+    @Get('/readiness')
+    @HealthCheck()
+    readiness() {
+        return this.health.check([
+            () => this.http.pingCheck('ethconnect', `${this.tokensService.baseUrl}/contracts`, basicAuth(this.tokensService.username, this.tokensService.password)),
+            () => this.http.pingCheck('ethconnect-contract', this.tokensService.instanceUrl, basicAuth(this.tokensService.username, this.tokensService.password)),
```

Comment:
Not sure how useful have both of these checks is since erc1155 won't startup without the contract deployed. Could use input from @awrichar on what a healthy ethconnect might be relative to erc1155.

Reply:
You're right that it'll just fail to startup if the contract isn't deployed - but I assume that's alright since the health APIs will just 404 in that case?

Seems like these health APIs will be more for detecting if something goes away mid-flight (like ethconnect is suddenly down or unreachable). These checks seem good for that. The first one may be redundant, but it's harmless to have in any case.

Reply:
Startup failure if the contract isn't deployed / available is not an issue and feels correct to me!

And you got it, this is in the event Ethconnect goes down during the tokens microservices lifetime and to indicate to K8s that it shouldn't retrieve traffic until Ethconnect comes back up.